# Question 4

In [1]:
using JuMP, Gurobi

minInvestment = [3 2 9 5 12 4]
maxInvestment = [27 12 35 15 46 18]
expectedRtn = [0.13 0.09 0.17 0.1 0.22 0.12]

totalInvestment = 80

n = length(expectedRtn)

m = Model(solver = GurobiSolver(OutputFlag = 0))

@variable(m, x[1:n], Bin)
@variable(m, y[1:n] >= 0) # amount of investment (million dollars)

@constraint(m, y .<= totalInvestment * x)
@constraint(m, sum{y[i] , i=1:n} <= totalInvestment)

for i=1:n
    @constraint(m, minInvestment[i] * x[i] <= y[i])
    #@constraint(m, y[i] * x[i] >= minInvestment[i])
    @constraint(m, y[i] * x[i] <= maxInvestment[i])
end

@constraint(m, y[5] <= y[2] + y[4] + y[6])
@constraint(m, x[3] <= x[6])

@objective(m, Max, sum{y[i] * expectedRtn[i], i=1:n})

status = solve(m)
println(status)
println("x= ", getvalue(x))
println("y= ", getvalue(y))
println("objective = ", getobjectivevalue(m))

Optimal
x= [-0.0, -0.0, 1.0, 1.0, 1.0, 1.0]
y= [0.0, 0.0, 35.0, 5.0, 22.5, 17.5]
objective = 13.5


We should invest option3 35 million, option4 5 million, option 5 22.5 million and option6 17.5 million.